packages/Algernon.package/ALGCore.class/instance/checkWorldBounds.st

```diff
@@ -0,0 +1,6 @@
+visibility
+checkWorldBounds
+
+	(self lastWorldBounds x > World bounds corner x
```

Comment:
This doesn't check for the current Position of Algernon at all. If I remember correctly, we wanted to preserve the Position of Algernon if it is still in View.

In this context it may also be reasonable to check if the Algernon bounds are negative, that is, if the window was moved left out of the image.

Reply:
As far as we know, we wanted to move algernon to the middle of the Screen when the screen has been made smaller and keep the position when the window gets bigger

Reply:
I'm not sure. @BenBals what do you think? Should we check if the Algernon is out of view due to the scaling down of the window or just recenter it no mater what ?

Reply:
We should only move Algernon if required.

Reply:
If I'm correct, that would also mean that we don't need a instance variable, making the computation stateless, which should be preferred.